src/utils/components/DiskModal/EditDiskModal.tsx

```diff
@@ -0,0 +1,183 @@
+import * as React from 'react';
+
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import TabModal from '@kubevirt-utils/components/TabModal/TabModal';
+import { getDataVolumeTemplates, getDisks, getVolumes } from '@kubevirt-utils/resources/vm';
+import { Form } from '@patternfly/react-core';
+
+import AccessMode from './DiskFormFields/AccessMode';
+import ApplyStorageProfileSettingsCheckbox from './DiskFormFields/ApplyStorageProfileSettingsCheckbox';
+import DetachHotplugDiskCheckbox from './DiskFormFields/DetachHotplugDiskCheckbox';
+import DiskInterfaceSelect from './DiskFormFields/DiskInterfaceSelect';
+import DiskSourceSizeInput from './DiskFormFields/DiskSizeInput/DiskSizeInput';
+import DiskSourceFormSelect from './DiskFormFields/DiskSourceFormSelect/DiskSourceFormSelect';
+import DiskTypeSelect from './DiskFormFields/DiskTypeSelect';
+import EnablePreallocationCheckbox from './DiskFormFields/EnablePreallocationCheckbox';
+import NameFormField from './DiskFormFields/NameFormField';
+import StorageClassSelect from './DiskFormFields/StorageClassSelect';
+import VolumeMode from './DiskFormFields/VolumeMode';
+import { DiskFormState, DiskSourceState } from './state/initialState';
+import { diskReducer, diskSourceReducer } from './state/reducers';
+import {
+  getDataVolumeFromState,
+  getDataVolumeTemplate,
+  getDiskFromState,
+  produceVMDisks,
+  requiresDataVolume,
+  updateVolume,
+} from './utils/helpers';
+
+type DiskModalProps = {
+  vm: V1VirtualMachine;
+  isOpen: boolean;
+  onClose: () => void;
+  headerText: string;
+  onSubmit: (updatedVM: V1VirtualMachine) => Promise<V1VirtualMachine | void>;
+  initialDiskState: DiskFormState;
+  initialDiskSourceState: DiskSourceState;
+};
+
+const EditDiskModal: React.FC<DiskModalProps> = ({
+  vm,
+  isOpen,
+  onClose,
+  headerText,
+  onSubmit,
+  initialDiskState,
+  initialDiskSourceState,
+}) => {
+  const [diskState, dispatchDiskState] = React.useReducer(diskReducer, initialDiskState);
+  const [diskSourceState, dispatchDiskSourceState] = React.useReducer(
+    diskSourceReducer,
+    initialDiskSourceState,
+  );
+  const sourceRequiresDataVolume = React.useMemo(
+    () => requiresDataVolume(diskState.diskSource),
+    [diskState.diskSource],
+  );
+
+  const updatedVirtualMachine: V1VirtualMachine = React.useMemo(() => {
```

Comment:
Can we split this function into smaller ones and move them into a utils file?

Reply:
fixed thanks :)